src/hdmf/build/manager.py

```diff
@@ -497,12 +497,51 @@ def __ischild(self, dtype):
 
     @staticmethod
     def __set_default_name(docval_args, default_name):
-        new_docval_args = []
-        for x in docval_args:
-            if x['name'] == 'name':
-                x['default'] = default_name
-            new_docval_args.append(x)
-        return new_docval_args
+        if default_name is not None:
+            for x in docval_args:
+                if x['name'] == 'name':
+                    x['default'] = default_name
```

Comment:
I added a test for this and broke a ton of other unrelated tests. It turns out that line 514 does not copy the dicts within the docval tuple, but copies references to the dicts, so the line above changes the 'name' docval arg for the base class. I'll fix this